src/index.tsx

```diff
@@ -34,31 +32,12 @@ function useLocationChange() {
     const dispatch = useDispatch();
 
     React.useEffect(() => {
-        if (location.pathname === VIEWER_PATHNAME) {
-            /**
-             * Gets called every time the app navigates to the Simularium page.
-             * There are 2 possible url types and 3 different states the viewer should be in
-             * when we get there:
-             * 1. Empty viewer, by clicking "launch viewer" or "load your own data" card; url: '/viewer'
-             * 2. Loading network file; url: '/viewer?trajFileName=FILE_NAME'
-             * 3. Loading a local file through the dropdown, will be reflected in the location state; url: '/viewer'
-             */
-            const state = store.getState();
-            const controller = getSimulariumController(state);
-            const simFile = getSimulariumFile(state);
-            // got here from the "load local file button" so the app is going to
-            // `/viewer`, but the loadFile action will take care of resetting state
-            // if the user clicks "Open"
-            if (location.state && location.state.localFile) {
-                return;
-            }
-            if (!location.search && controller && simFile.name) {
-                // going to /viewer, clear out any existing files
+        if (location.pathname !== VIEWER_PATHNAME) {
+            batch(() => {
+                // if we're navigating away from the viewer, stop playing and reset state
+                dispatch(setIsPlaying(false));
                 dispatch(clearSimulariumFile({ newFile: false }));
-            }
-        } else {
-            // if we're navigating away from the viewer, stop playing
-            dispatch(setIsPlaying(false));
+            });
```

Comment:
Essentially, I'm proposing clearing out the state when leaving the viewer page instead of when arriving at the viewer page.

Reply:
this seems a lot cleaner. I had to think a bit about if this misses anything, but it seems good? Have you tested a lot of different navigation combos? Like loading a file, going back to home, loading a file from home, etc?

Reply:
Cool. Yeah, I haven't been able to break it yet :)